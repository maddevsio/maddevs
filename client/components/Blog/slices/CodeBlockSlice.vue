<template>
  <div class='post-part single'>
    <prism-editor
      class="code-sample"
      v-model="code"
      :highlight="highlighter"
      readonly
    />
  </div>
</template>

<script>
import { PrismEditor } from 'vue-prism-editor';
import 'vue-prism-editor/dist/prismeditor.min.css';

import 'prismjs';
import {highlight} from 'prismjs';
import 'prismjs/components/prism-clike';
import 'prismjs/components/prism-javascript';
import 'prismjs/components/prism-css.min';
import 'prismjs/themes/prism.css';

export default {
  name: 'CodeblockSlice',
  props: {
    slice: {
      type: Object,
      required: true
    }
  },
  data(){
    return {
      code: this.slice.primary.code.reduce((prev, element) => prev + element.text, '')
    };
  },
  components: {
    PrismEditor
  },
  methods: {
    highlighter(code){
      return highlight(code, Prism.languages[this.slice.primary.language[0].text], this.slice.primary.language[0].text);
    }
  }
};
</script>

<style scoped lang="scss">
  @import '../../../assets/styles/_vars';
  @import url('https://fonts.googleapis.com/css2?family=IBM+Plex+Mono&display=swap');

  .code-sample {
    background-color: whitesmoke;
    padding: 25px;
    line-height: 22px;
    font-size: 17px;
    font-family: 'IBM Plex Mono', monospace;
  }
</style>

