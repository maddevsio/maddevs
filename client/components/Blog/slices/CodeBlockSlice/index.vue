<template>
  <div class="post-part single">
    <prism-editor v-model="code" :highlight="highlighter" class="code-sample" readonly />
  </div>
</template>

<script>
import { PrismEditor } from 'vue-prism-editor'
import 'vue-prism-editor/dist/prismeditor.min.css'

import 'prismjs'
import Prism from 'prismjs/components/prism-core.min'
import '@/components/Blog/slices/CodeBlockSlice/highlights'

export default {
  name: 'CodeblockSlice',
  components: {
    PrismEditor,
  },

  props: {
    slice: {
      type: Object,
      required: true,
    },
  },

  data() {
    return {
      code: '',
    }
  },

  mounted() {
    this.code = this.slice.primary.code.reduce((prev, element) => prev + element.text, '')
  },

  methods: {
    highlighter(code) {
      return Prism.highlight(code, Prism.languages[this.slice.primary.language], this.slice.primary.language)
    },
  },
}
</script>

<style scoped lang="scss">
@import '../../../../assets/styles/_vars';

.code-sample {
  line-height: 22px;
  font-size: 17px;
  font-family: 'IBM Plex Mono', monospace;
  margin: 24px 0;

  /deep/ .prism-editor__container {
    background-color: whitesmoke;
  }

  /deep/ .prism-editor__container,
  /deep/ .prism-editor__textarea {
    padding: 25px;
  }
}

@media only screen and (max-width: 1024px) {
  .code-sample {
    font-size: 16px;
    line-height: 24px;
  }
}
</style>

