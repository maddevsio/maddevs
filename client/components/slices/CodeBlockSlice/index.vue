<template>
  <div class="post-part single">
    <PrismEditor
      v-model="code"
      :highlight="highlighter"
      class="code-sample"
      readonly
    />
  </div>
</template>

<script>
import { PrismEditor } from 'vue-prism-editor'
import 'vue-prism-editor/dist/prismeditor.min.css'

import 'prismjs'
import Prism from 'prismjs/components/prism-core.min'
import '@/components/slices/CodeBlockSlice/highlights'

export default {
  name: 'CodeblockSlice',
  components: {
    PrismEditor,
  },

  props: {
    slice: {
      type: Object,
      required: true,
    },
  },

  data() {
    return {
      code: '',
    }
  },

  mounted() {
    this.code = this.slice.primary.code.reduce((prev, element) => prev + element.text, '')
  },

  methods: {
    highlighter(code) {
      return Prism.highlight(code, Prism.languages[this.slice.primary.language], this.slice.primary.language)
    },
  },
}
</script>

<style scoped lang="scss">
.code-sample {
  font-family: 'IBM Plex Mono', monospace;
  font-size: 14px;
  font-weight: 400;
  line-height: 22px;
  margin: 24px 0;

  /deep/ .prism-editor__container {
    background-color: whitesmoke;
  }

  /deep/ .prism-editor__container,
  /deep/ .prism-editor__textarea {
    padding: 25px;
  }
}

@media only screen and (max-width: 1024px) {
  .code-sample {
    font-size: 16px;
    line-height: 24px;
  }
}
</style>
