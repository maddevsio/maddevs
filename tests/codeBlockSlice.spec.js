import {
  mount
} from '@vue/test-utils';
import CodeBlockSlice from '../client/components/Blog/slices/CodeBlockSlice';

describe('Code block slice component', () => {
  let wrapper;

  const slice = {
    items: [{}],
    primary: {
      code: [{
        spans: [],
        text: 'getAllHeadings() {\n  this.document.body.forEach(listItem => {\n    if(listItem.primary.text !== undefined) {\n      if(listItem.primary.text[0].type === \'heading1\') {\n        this.headingsList.push({\n          textContent: listItem.primary.text[0].text,\n          headingName: listItem.primary.text[0].text.toLowerCase().replace(/\s/g , \'-\')\n        });\n      }\n    }\n  });',
        type: 'paragraph'
      }],
      language: 'javascript'
    },
    slice_label: null,
    slice_type: 'codeblock'
  };

  beforeEach(() => {
    wrapper = mount(CodeBlockSlice, {
      propsData: {slice}
    });
  });

  test('Is a Vue instance', () => {
    expect(wrapper.exists()).toBeTruthy();
  });

  test('Renders correctly', () => {
    expect(wrapper.element).toMatchSnapshot();
  });
});
