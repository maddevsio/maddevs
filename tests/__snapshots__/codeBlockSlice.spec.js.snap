// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Code block slice component Renders correctly 1`] = `
<div
  class="post-part single"
>
  <div
    class="prism-editor-wrapper code-sample"
  >
    <div
      class="prism-editor__container"
    >
      <textarea
        autocapitalize="off"
        autocomplete="off"
        autocorrect="off"
        class="prism-editor__textarea"
        data-gramm="false"
        data-testid="textarea"
        placeholder=""
        readonly="readonly"
        spellcheck="false"
      />
      <pre
        class="prism-editor__editor"
        data-testid="preview"
      >
        <span
          class="token function"
        >
          getAllHeadings
        </span>
        <span
          class="token punctuation"
        >
          (
        </span>
        <span
          class="token punctuation"
        >
          )
        </span>
         
        <span
          class="token punctuation"
        >
          {
        </span>
        
  
        <span
          class="token keyword"
        >
          this
        </span>
        <span
          class="token punctuation"
        >
          .
        </span>
        document
        <span
          class="token punctuation"
        >
          .
        </span>
        body
        <span
          class="token punctuation"
        >
          .
        </span>
        <span
          class="token function"
        >
          forEach
        </span>
        <span
          class="token punctuation"
        >
          (
        </span>
        <span
          class="token parameter"
        >
          listItem
        </span>
         
        <span
          class="token operator"
        >
          =&gt;
        </span>
         
        <span
          class="token punctuation"
        >
          {
        </span>
        
    
        <span
          class="token keyword"
        >
          if
        </span>
        <span
          class="token punctuation"
        >
          (
        </span>
        listItem
        <span
          class="token punctuation"
        >
          .
        </span>
        primary
        <span
          class="token punctuation"
        >
          .
        </span>
        text 
        <span
          class="token operator"
        >
          !==
        </span>
         
        <span
          class="token keyword"
        >
          undefined
        </span>
        <span
          class="token punctuation"
        >
          )
        </span>
         
        <span
          class="token punctuation"
        >
          {
        </span>
        
      
        <span
          class="token keyword"
        >
          if
        </span>
        <span
          class="token punctuation"
        >
          (
        </span>
        listItem
        <span
          class="token punctuation"
        >
          .
        </span>
        primary
        <span
          class="token punctuation"
        >
          .
        </span>
        text
        <span
          class="token punctuation"
        >
          [
        </span>
        <span
          class="token number"
        >
          0
        </span>
        <span
          class="token punctuation"
        >
          ]
        </span>
        <span
          class="token punctuation"
        >
          .
        </span>
        type 
        <span
          class="token operator"
        >
          ===
        </span>
         
        <span
          class="token string"
        >
          'heading1'
        </span>
        <span
          class="token punctuation"
        >
          )
        </span>
         
        <span
          class="token punctuation"
        >
          {
        </span>
        
        
        <span
          class="token keyword"
        >
          this
        </span>
        <span
          class="token punctuation"
        >
          .
        </span>
        headingsList
        <span
          class="token punctuation"
        >
          .
        </span>
        <span
          class="token function"
        >
          push
        </span>
        <span
          class="token punctuation"
        >
          (
        </span>
        <span
          class="token punctuation"
        >
          {
        </span>
        
          textContent
        <span
          class="token operator"
        >
          :
        </span>
         listItem
        <span
          class="token punctuation"
        >
          .
        </span>
        primary
        <span
          class="token punctuation"
        >
          .
        </span>
        text
        <span
          class="token punctuation"
        >
          [
        </span>
        <span
          class="token number"
        >
          0
        </span>
        <span
          class="token punctuation"
        >
          ]
        </span>
        <span
          class="token punctuation"
        >
          .
        </span>
        text
        <span
          class="token punctuation"
        >
          ,
        </span>
        
          headingName
        <span
          class="token operator"
        >
          :
        </span>
         listItem
        <span
          class="token punctuation"
        >
          .
        </span>
        primary
        <span
          class="token punctuation"
        >
          .
        </span>
        text
        <span
          class="token punctuation"
        >
          [
        </span>
        <span
          class="token number"
        >
          0
        </span>
        <span
          class="token punctuation"
        >
          ]
        </span>
        <span
          class="token punctuation"
        >
          .
        </span>
        text
        <span
          class="token punctuation"
        >
          .
        </span>
        <span
          class="token function"
        >
          toLowerCase
        </span>
        <span
          class="token punctuation"
        >
          (
        </span>
        <span
          class="token punctuation"
        >
          )
        </span>
        <span
          class="token punctuation"
        >
          .
        </span>
        <span
          class="token function"
        >
          replace
        </span>
        <span
          class="token punctuation"
        >
          (
        </span>
        <span
          class="token regex"
        >
          <span
            class="token regex-delimiter"
          >
            /
          </span>
          <span
            class="token regex-source language-regex"
          >
            s
          </span>
          <span
            class="token regex-delimiter"
          >
            /
          </span>
          <span
            class="token regex-flags"
          >
            g
          </span>
        </span>
         
        <span
          class="token punctuation"
        >
          ,
        </span>
         
        <span
          class="token string"
        >
          '-'
        </span>
        <span
          class="token punctuation"
        >
          )
        </span>
        
        
        <span
          class="token punctuation"
        >
          }
        </span>
        <span
          class="token punctuation"
        >
          )
        </span>
        <span
          class="token punctuation"
        >
          ;
        </span>
        
      
        <span
          class="token punctuation"
        >
          }
        </span>
        
    
        <span
          class="token punctuation"
        >
          }
        </span>
        
  
        <span
          class="token punctuation"
        >
          }
        </span>
        <span
          class="token punctuation"
        >
          )
        </span>
        <span
          class="token punctuation"
        >
          ;
        </span>
        <br />
      </pre>
    </div>
  </div>
</div>
`;
